### PR TITLE
CATROID-609 Change Jumping Sumo setting to CheckBoxPreference

### PR DESCRIPTION
**Pull request by mandelbrot2357**
Eliminated unnecessary secondary settings screen for Jumping Sumo extension

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [x] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: catroid/src/main/res/xml/preferences.xml [hunk 1]
```diff
@@ -64,12 +64,13 @@
 
     </PreferenceScreen>
 
-    <PreferenceScreen
-            android:key="setting_parrot_jumping_sumo_bricks"
-            android:summary="@string/preference_description_jumpingsumo_bricks"
-            android:title="@string/preference_title_enable_jumpingsumo_bricks" >
+    <CheckBoxPreference
+        android:defaultValue="@string/FEATURE_ARDUINO_PREFERENCES_ENABLED"
```
Comment (PlantinumX) [inline catroid/src/main/res/xml/preferences.xml:68]: The jumping sumo feature should not be enabled by the arduino flavor itself. so let it per default false.

**Comment by PlantinumX** [comment]
@wslany  what do you think ?

**Comment by ndrnour** [comment]
**just sharing my opinion:** 
you're going to revert this commit someday in the future once someone starts implementing new configurations like:
* set a name to Jumping Sumo
* change resolution or aspect ratio of taken pictures or recorded videos
* Modifying the color and luminosity of the eyes
.... etc

**Comment by wslany** [comment]
> **just sharing my opinion:**
> you're going to revert this commit someday in the future once someone starts implementing new configurations like:
> 
> * set a name to Jumping Sumo
> * change resolution or aspect ratio of taken pictures or recorded videos
> * Modifying the color and luminosity of the eyes
>   .... etc

Sure, but it is not sure that we will ever do all those things, especially since we are now working on a more generic solution for all kind of drones and robots. So, after some time the Jumping Sumo drone support may become obsolete anyway. The product is from 2017. 

The YAGNI principle (https://www.martinfowler.com/bliki/Yagni.html ) we are using proposes to not take into account anything that is not going to be immediately implemented. I.e., possible future features need not to be considered. And also, even if the features you mention ever come, it will not be complex to implement the additional dialog, since we have examples for the other settings. It might even be a good occasion to refactor to Kotlin and also to eliminate code duplication. 

Having said that, the change is also not very important, as we do not really have currently many users using this drone. This could change if we ever create a special flavor for the drone, but in that case it would anyway be always be turned on, so there would be no need for the checkbox at all.

**Comment by PlantinumX** [comment]
Hi @mandelbrot2357  pls resolve conflics than I will merge it thx

**Comment by wslany** [comment]
Thanks!

**Pull request merged**